kotlinx-nodejs/src/url.url.module_node.kt

```diff
@@ -17,74 +17,12 @@ import org.w3c.notifications.*
 import org.w3c.performance.*
 import org.w3c.workers.*
 import org.w3c.xhr.*
-import querystring.ParsedUrlQuery
-
-external interface UrlObject {
-    var auth: String?
-        get() = definedExternally
-        set(value) = definedExternally
-    var hash: String?
-        get() = definedExternally
-        set(value) = definedExternally
-    var host: String?
-        get() = definedExternally
-        set(value) = definedExternally
-    var hostname: String?
-        get() = definedExternally
-        set(value) = definedExternally
-    var href: String?
-        get() = definedExternally
-        set(value) = definedExternally
-    var pathname: String?
-        get() = definedExternally
-        set(value) = definedExternally
-    var protocol: String?
-        get() = definedExternally
-        set(value) = definedExternally
-    var search: String?
-        get() = definedExternally
-        set(value) = definedExternally
-    var slashes: Boolean?
-        get() = definedExternally
-        set(value) = definedExternally
-    var port: dynamic /* String? | Number? */
-        get() = definedExternally
-        set(value) = definedExternally
-    var query: dynamic /* String? | ParsedUrlQueryInput? */
-        get() = definedExternally
-        set(value) = definedExternally
-}
-
-external interface Url {
-    var auth: String?
-    var hash: String?
-    var host: String?
-    var hostname: String?
-    var href: String
-    var path: String?
-    var pathname: String?
-    var protocol: String?
-    var search: String?
-    var slashes: Boolean?
-    var port: String?
-    var query: dynamic /* String? | ParsedUrlQuery? */
-        get() = definedExternally
-        set(value) = definedExternally
-}
-
-external interface UrlWithParsedQuery : Url {
-    override var query: ParsedUrlQuery
-}
-
-external interface UrlWithStringQuery : Url {
-    override var query: String?
-}
 
 external fun parse(urlStr: String): UrlWithStringQuery
 
 external fun parse(urlStr: String, parseQueryString: Boolean?, slashesDenoteHost: Boolean = definedExternally): UrlWithStringQuery
 
-external fun parse(urlStr: String, parseQueryString: Boolean, slashesDenoteHost: Boolean = definedExternally): Url
+external fun parse(urlStr: String, parseQueryString: Boolean, slashesDenoteHost: Boolean = definedExternally): dynamic /* UrlWithParsedQuery | Url */
```

Comment:
please check it

Reply:
here's what's basically happening:
```ts
interface Url {}
interface UrlWithParsedQuery extends Url {}

function parse(urlStr: string, parseQueryString: true, slashesDenoteHost?: boolean): UrlWithParsedQuery;
function parse(urlStr: string, parseQueryString: boolean, slashesDenoteHost?: boolean): Url;
```

used to be translated to:
```ts
external interface Url
external interface UrlWithParsedQuery : Url
external fun parse(urlStr: String, parseQueryString: Boolean, slashesDenoteHost: Boolean = definedExternally): UrlWithParsedQuery
external fun parse(urlStr: String, parseQueryString: Boolean, slashesDenoteHost: Boolean = definedExternally): Url
```

which is a conflicting overload - the policy for such overloads was to start deleting overloads from above till there's no conflicts, thus ` parse(urlStr: String, parseQueryString: Boolean, slashesDenoteHost: Boolean = definedExternally): Url` remained. 

since then we've decided to just merge functions in such cases - and create a unit return type, so pretty much following happens:
`parse(urlStr: String, parseQueryString: Boolean, slashesDenoteHost: Boolean = definedExternally): Url | UrlWithParsedQuery` => `parse(urlStr: String, parseQueryString: Boolean, slashesDenoteHost: Boolean = definedExternally): dynamic`.

There's obviously room for improvement - we can use the most common ancestor in that case (so we'll have Url as return type) -but for this we need to design and test this feature. As of now return types in unions resolved as dynamic,